core/common/src/grpc_interceptor.rs

```diff
@@ -113,22 +97,29 @@ pub struct GrpcInterceptorService<S> {
 
 impl<S> GrpcInterceptorService<S> {
     /// Retrieve the gRPC service name and method name from a URI.
+    /// If it cannot succesfully be parsed, then an empty service name and/or method name will be returned.
     ///
     /// * `uri` - The uri used for the gRPC call.
     fn retrieve_grpc_names_from_uri(uri: &Uri) -> (String, String) {
-        let uri_string = uri.to_string();
-        let uri_parts: Vec<&str> = uri_string.split('/').collect();
         let mut service_name = String::new();
         let mut method_name = String::new();
-        if uri_parts.len() == 5 {
-            method_name = uri_parts[4].to_string();
-            let qualified_service_name = uri_parts[3].to_string();
-            let qualified_service_name_parts: Vec<&str> =
-                qualified_service_name.split('.').collect();
-            if qualified_service_name_parts.len() == 2 {
-                service_name = qualified_service_name_parts[1].to_string();
+        // A gRPC URI path looks like this "/invehicle_digital_twin.InvehicleDigitalTwin/FindById".
+        match Regex::new(r"^/[^/\.]+\.([^/]+)/(.+)$") {
+            Ok(regex_pattern) => {
+                if let Some(caps) = regex_pattern.captures(uri.path()) {
+                    // Note: caps.get(0) represents the entire string that matched.
+                    //       In the earlier gRPC URI path example it would be
+                    //       "/invehicle_digital_twin.InvehicleDigitalTwin/FindById".
+                    //       caps.get(1) and caps.get(2) represent the sub-parts that matched.
+                    if caps.len() == 3 {
```

Comment:
Consider assigning a constant to `3`

Reply:
The number 3 is associated with the regex pattern.  If the pattern changes, then this will change too.  No need for a constant.